File: Sources/CountriesFlagsObjC/source/Countries/UIImage+GuineaBissau.m
//
//  UIImage+GuineaBissau.m
//  CountriesFlags
//

#import "Countries/UIImage+GuineaBissau.h"

#import "Extensions/UIBezierPath+Star.h"

@implementation UIImage (GuineaBissau)

+ (UIImage * _Nullable)flagGuineaBissauWithSize:(CGSize)size {
    UIColor* colorRed = [UIColor colorWithRed:206.0f/255.0f
                                        green:17.0f/255.0f
                                         blue:38.0f/255.0f
                                        alpha:1.0f];
    UIColor* colorYellow = [UIColor colorWithRed:252.0f/255.0f
                                           green:209.0f/255.0f
                                            blue:22.0f/255.0f
                                           alpha:1.0f];
    UIColor* colorGreen = [UIColor colorWithRed:0.0f
                                          green:158.0f/255.0f
                                           blue:73.0f/255.0f
                                          alpha:1.0f];
    UIColor* colorBlack = [UIColor blackColor];
    
    UIGraphicsBeginImageContextWithOptions(size, NO, 0.0f);
    
    [colorYellow setFill];
    UIRectFill(CGRectMake(0.0f, 0.0f, size.width, size.height));
    
    [colorGreen setFill];
    UIRectFill(CGRectMake(0.0f, size.height / 2.0, size.width, size.height / 2.0));
    
    [colorGreen setFill];
    UIRectFill(CGRectMake(0.0f, size.height / 2.0, size.width, size.height / 2.0));
    
    [colorRed setFill];
    UIRectFill(CGRectMake(0.0f, 0.0f, size.width / 3.0f, size.height));
    
    [colorBlack setFill];
    [[UIBezierPath starWithSize:CGSizeMake(size.width / 3.0f, size.height)
                         radius:size.height / 3.0f / 2.0f
                           type:FlagStarTypeFivePointed] fill];
    
    UIImage* image = UIGraphicsGetImageFromCurrentImageContext();
    UIGraphicsEndImageContext();
    
    return image;
}

@end

File: Sources/CountriesFlags/Countries/UIImage+GuineaBissau.swift
//
//  UIImage+GuineaBissau.swift
//  CountriesFlags
//

#if canImport(UIKit)

import UIKit

extension UIImage {
    /**
     [wikipedia]: https://en.wikipedia.org/wiki/Flag_of_Guinea-Bissau "wikipedia"
     
     Generates the flag of Guinea-Bissau.
     For more information see [wikipedia].
     
     - Parameters:
        - size: image size.
     
     - Returns: Returns the flag of Guinea-Bissau.
     */
    internal static func flagGuineaBissau(with size: CGSize) -> UIImage? {
        let colorRed = UIColor(red: 206.0 / 255.0,
                               green: 17.0 / 255.0,
                               blue: 38.0 / 255.0,
                               alpha: 1.0)
        let colorYellow = UIColor(red: 252.0 / 255.0,
                                  green: 209.0 / 255.0,
                                  blue: 22.0 / 255.0,
                                  alpha: 1.0)
        let colorGreen = UIColor(red: 0.0,
                                 green: 158.0 / 255.0,
                                 blue: 73.0 / 255.0,
                                 alpha: 1.0)
        let colorBlack = UIColor.black

        UIGraphicsBeginImageContextWithOptions(size, false, 0.0)

        colorYellow.setFill()
        UIRectFill(CGRect(origin: .zero, size: size))

        colorGreen.setFill()
        UIRectFill(CGRect(origin: CGPoint(x: 0.0, y: size.height / 2.0),
                          size: CGSize(width: size.width, height: size.height / 2.0)))

        colorRed.setFill()
        UIRectFill(CGRect(origin: CGPoint(x: 0.0, y: 0.0),
                          size: CGSize(width: size.width / 3.0, height: size.height)))

        colorBlack.setFill()
        let radius = CGFloat(size.height / 3.0) / 2.0
        UIBezierPath.star(at: CGPoint(x: size.width / 6.0, y: size.height / 2.0),
                          radius: radius).fill()

        let image = UIGraphicsGetImageFromCurrentImageContext()
        UIGraphicsEndImageContext()

        return image
    }
}

#endif
